### PR TITLE
feat(extract): add PR component extraction and fix flaky git tests

### LINKED ISSUE #169: [M5] Add PR component extraction commands
### Context
**PRD:** `docs/project/PRD/active/PRD-phase-11-metadata-extraction.md` — Milestone 5

Teams need to understand architectural impact of PRs without extracting the entire codebase. This enables CI integration for PR comments showing added/modified/removed components.

### Traceability
- **PRD Section:** M5-D5.1 (--pr command), M5-D5.2 (--files command), M5-D5.3 (Output formats), M5-D5.4 (CLI docs)
- **Success Criteria:** SC-5 (riviere extract --pr command available)

### Acceptance Criteria
- [ ] `riviere extract --pr` detects changed files vs base branch
- [ ] `riviere extract --pr --base <branch>` overrides default base branch
- [ ] `riviere extract --files <file1> <file2>` extracts from specific files
- [ ] `--format json` outputs JSON (default)
- [ ] `--format markdown` outputs PR-comment-ready markdown
- [ ] Base branch detection: `git symbolic-ref refs/remotes/origin/HEAD`, fallback to `main`
- [ ] Handles detached HEAD by using `HEAD~1` as base
- [ ] Handles uncommitted changes: includes staged files, warns about unstaged
- [ ] No changed TypeScript files → empty array, exit 0
- [ ] CLI docs updated with PR extraction examples

### Edge Case Scenarios
- Not a git repo → exit 3: `Error: [GIT_ERROR] NOT_A_REPOSITORY`
- Git not installed → exit 3: `Error: [GIT_ERROR] GIT_NOT_FOUND`
- Detached HEAD → use `HEAD~1` as base
- Uncommitted staged changes → include in extraction, warn about unstaged
- No TypeScript files changed → empty components array, exit 0
- Base branch doesn't exist → exit 3 with clear error
- File in --files list doesn't exist → exit 3 with error listing missing file
- Changed file outside any module path → skip silently (not an error)

### Implementation Guidelines
- **Location:** `packages/riviere-cli/src/commands/extract/`
- Use Node.js `child_process.execSync` for git commands (no additional git library)
- Git detection order: repo check → HEAD check → uncommitted check → diff
- Markdown format follows PRD Section 3.7 (Added/Modified/Removed sections)
- Changed file detection: `git diff --name-only <base>...HEAD`
- Filter to .ts/.tsx files only

### Embedded Reasoning
**Why:** PR-scoped extraction enables CI integration, showing architectural changes in PR comments without full codebase scan.
**What:** --pr and --files flags, JSON and Markdown output formats.
**How:** Shell out to git CLI for file detection, filter to TypeScript, run standard extraction on subset.

### Testing Strategy
- **Unit:** Git command execution and parsing
- **Unit:** Base branch detection logic
- **Unit:** Markdown output formatting
- **Integration:** CLI invocation in test git repo
- **Edge Cases:** All scenarios from Edge Case Scenarios section (use temp git repos)

### Dependencies
Depends on #166 (CLI infrastructure with flags)

### Verification
```bash
pnpm nx build riviere-cli
./packages/riviere-cli/dist/bin.js extract --help | grep -E '(--pr|--files|--base|--format)'
# Test in actual repo
./packages/riviere-cli/dist/bin.js extract --pr --dry-run
```

### PR DESCRIPTION
**Pull request by NTCoding**
Closes #169

### Context
**PRD:** `docs/project/PRD/active/PRD-phase-11-metadata-extraction.md` — Milestone 5

Teams need to understand architectural impact of PRs without extracting the entire codebase. This enables CI integration for PR comments showing added/modified/removed components.

### Traceability
- **PRD Section:** M5-D5.1 (--pr command), M5-D5.2 (--files command), M5-D5.3 (Output formats), M5-D5.4 (CLI docs)
- **Success Criteria:** SC-5 (riviere extract --pr command available)

### Acceptance Criteria
- [ ] `riviere extract --pr` detects changed files vs base branch
- [ ] `riviere extract --pr --base <branch>` overrides default base branch
- [ ] `riviere extract --files <file1> <file2>` extracts from specific files
- [ ] `--format json` outputs JSON (default)
- [ ] `--format markdown` outputs PR-comment-ready markdown
- [ ] Base branch detection: `git symbolic-ref refs/remotes/origin/HEAD`, fallback to `main`
- [ ] Handles detached HEAD by using `HEAD~1` as base
- [ ] Handles uncommitted changes: includes staged files, warns about unstaged
- [ ] No changed TypeScript files → empty array, exit 0
- [ ] CLI docs updated with PR extraction examples

### Edge Case Scenarios
- Not a git repo → exit 3: `Error: [GIT_ERROR] NOT_A_REPOSITORY`
- Git not installed → exit 3: `Error: [GIT_ERROR] GIT_NOT_FOUND`
- Detached HEAD → use `HEAD~1` as base
- Uncommitted staged changes → include in extraction, warn about unstaged
- No TypeScript files changed → empty components array, exit 0
- Base branch doesn't exist → exit 3 with clear error
- File in --files list doesn't exist → exit 3 with error listing missing file
- Changed file outside any module path → skip silently (not an error)

### Implementation Guidelines
- **Location:** `packages/riviere-cli/src/commands/extract/`
- Use Node.js `child_process.execSync` for git commands (no additional git library)
- Git detection order: repo check → HEAD check → uncommitted check → diff
- Markdown format follows PRD Section 3.7 (Added/Modified/Removed sections)
- Changed file detection: `git diff --name-only <base>...HEAD`
- Filter to .ts/.tsx files only

### Embedded Reasoning
**Why:** PR-scoped extraction enables CI integration, showing architectural changes in PR comments without full codebase scan.
**What:** --pr and --files flags, JSON and Markdown output formats.
**How:** Shell out to git CLI for file detection, filter to TypeScript, run standard extraction on subset.

### Testing Strategy
- **Unit:** Git command execution and parsing
- **Unit:** Base branch detection logic
- **Unit:** Markdown output formatting
- **Integration:** CLI invocation in test git repo
- **Edge Cases:** All scenarios from Edge Case Scenarios section (use temp git repos)

### Dependencies
Depends on #166 (CLI infrastructure with flags)

### Verification
```bash
pnpm nx build riviere-cli
./packages/riviere-cli/dist/bin.js extract --help | grep -E '(--pr|--files|--base|--format)'
# Test in actual repo
./packages/riviere-cli/dist/bin.js extract --pr --dry-run
```

## Summary by CodeRabbit

* **New Features**
  * Pull-request extraction via --pr with optional --base; explicit file selection via --files; markdown PR summaries via --format markdown

* **Improvements**
  * Better Git-related error reporting (not-a-repo / git not found) and warnings for unstaged/untracked TypeScript files
  * Mutual-exclusion and format validation for new flags
  * More robust code-review verdict parsing across multi-line outputs

* **Tests**
  * Extensive new test coverage for PR flows, git change detection, source-file filtering, and Markdown formatting

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: packages/riviere-cli/src/features/extract/commands/filter-source-files.spec.ts [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+      const dir = makeTempDir()
+      process.chdir(dir)
+      mockDetectChanged.mockImplementation(() => {
+        throw new GitError('NOT_A_REPOSITORY', 'Run from within a git repository.')
+      })
+
+      expect(() => filterSourceFiles([], { pr: true })).toThrow(
+        expect.objectContaining({ filterErrorKind: 'GIT_ERROR' }),
+      )
+    })
```
[the remark this answers is not in the document]
Comment (NTCoding) [reply packages/riviere-cli/src/features/extract/commands/filter-source-files.spec.ts]: @coderabbitai you are right here, claude is wrong. they are new tests and they should follow conventions.
